prottp.go

```diff
@@ -151,39 +185,41 @@ func wrapMethod(service interface{}, m grpc.MethodDesc) http.Handler {
 			}
 			return
 		}
-		writeMessage(isJson, resp.(proto.Message), http.StatusOK, w)
+		WriteMessage(statusCode, resp.(proto.Message), w, r)
 	})
 }
 
-func writeMessage(isJson bool, msg proto.Message, statusCode int, w http.ResponseWriter) {
+// WriteMessage is exported to be used for middleware to return proto message
+func WriteMessage(statusCode int, msg proto.Message, w http.ResponseWriter, r *http.Request) {
 	var err error
+	var isJSON = isMimeTypeJSON(w.Header().Get("Content-Type"))
+	if w.Header().Get("Content-Type") == "" {
+		isJSON = shouldReturnJSON(r)
```

Comment:
The way you use these conditionals seems a bit weird. I don't really follow why there are several checks about the content type header and `isJson`.

Reply:
```go
var isJSON = isMimeTypeJSON(w.Header().Get("Content-Type"))
if w.Header().Get("Content-Type") == "" {
```

 is for other middleware can set the Content-Type, in that case, we use their settings

Reply:
situations:

1. Middleware set the ContentType by using `w.Header().Set()` to JSON, prottp should return JSON
2. HTTP Header Accept is JSON, prottp should return JSON
3. HTTP Request Content-Type is JSON, prottp should return JSON